.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -104,95 +119,265 @@ jobs:
-          SOURCE_DIGEST: ${{ inputs.source-digest || github.ref }}
+      - name: Generate the builder binary
+        id: generate-builder
         run: |
-          # We could either:
-          #   1. Run a simple script to generate the SLSA recipe
-          #   2. Create a simple action to generate the SLSA recipe
-          #   3. Publish a binary on GH actions to generate the SLSA
-          #      recipe.
-          echo "running fake slsa recipe generator"
-          echo "recipe=path/to/recipe" >> $GITHUB_OUTPUT
+          echo "fetches the builder binary and uploads it"
+          echo "sha256=5b36fa157544ece32dba62c14480d85322b86918032ca5f40b87f5d1dc386d39" >> $GITHUB_OUTPUT
 
   ###################################################################
   #                                                                 #
-  #                   Generate builder binary                       #
+  #     Invoke the builder binary to create a buildDefinition       #
   #                                                                 #
   ###################################################################
-  generate-builder:
-    # TODO: Should this step update the SLSA predicate with the
-    # builder dependency?
+  generate-build-definition:
+    # TODO: Use the builder binary.
     outputs:
-      go-builder-sha256: ${{ steps.generate.outputs.sha256 }}
+      build-definition-name: ${{ steps.generate.outputs.build-definition-name }}
+      build-definition-sha256: ${{ steps.upload.outputs.sha256 }}
     runs-on: ubuntu-latest
+    needs: [detect-env, verify-builder-image]
     steps:
-      - name: Generate the builder binary
+      - name: Generate build definition
+        id: generate
+        # These are the inputs, it may be with: for an action or
+        # specified with these env vars.
+        env:
+          BUILDER_IMAGE: ${{ inputs.builder-image }}
+          BUILDER_DIGEST: ${{ inputs.builder-digest }}
+          BUILDER_VERIFIED: ${{ needs.verify-builder-image.outputs.verified }}
+          CONFIG_PATH: ${{ inputs.config-path }}
+          OUTPUT: ${{ inputs.builder-output-path }}
+          # Either the provided source repository or the caller's GITHUB_REPOSITORY
+          SOURCE_REPOSITORY: ${{ inputs.source-repository || github.repository }}
+          # Either the provided digest, or the caller's GITHUB_REF
+          SOURCE_DIGEST: ${{ inputs.source-digest || github.ref }}
         run: |
-          echo "fetches the builder binary and uploads it"
+          # Run builder binary and produce buildDefinition
+          # TODO: Insert verified builder image provenance into resolvedDep
+          echo "run resolution and output buildDefinition!"
+
+          SOURCE_ALG=${SOURCE_DIGEST%%:*}
+          SOURCE_DIGEST=${SOURCE_DIGEST#*:}
+
+          IMAGE_ALG=${BUILDER_DIGEST%%:*}
+          IMAGE_DIGEST=${BUILDER_DIGEST#*:}
+
+          cat <<EOF >DATA
+          {
```

Comment:
The JSON format is not quite right, I think. The one I get from the Go structs is https://github.com/slsa-framework/slsa-github-generator/blob/c281078c3a9fe1d76451cca2f3605182d915d192/internal/builders/docker/testdata/build-definition.json.

Reply:
Oops, I think ou're right: the source/builderImage needed to be inlined in artifacts. I'll send a quick correction PR